pkg/virt-operator/resource/apply/rbac.go

```diff
@@ -263,25 +266,75 @@ func arePolicyRulesEqual(role1 runtime.Object, role2 runtime.Object) (equal bool
 
 		return true
 	}
+	changeExistingPolicyRulesByRequired := func(existing, required *[]rbacv1.PolicyRule) (modified bool) {
+		if !arePolicyRulesEqual(*existing, *required) {
+			*existing = *required
+			return true
+		}
+		return false
+	}
+	changeExistingSubjectsByRequired := func(existingSubjects, requiredSubjects *[]rbacv1.Subject) bool {
+		modified := false
+		if len(*existingSubjects) != len(*requiredSubjects) {
+			*existingSubjects = *requiredSubjects
+			return false
+		}
 
-	switch role1.(type) {
+		for _, existingSubject := range *existingSubjects {
```

Comment:
instead of a nested loop which has N2 complexity I would sort and then compare, but the list should also be relatively short so I guess it doesn't really matter :shrug:

Reply:
Exactly what I thought.
IMHO, when performance overhead is negligible (i.e. if we have very small arrays that are not supposed to ever grow in a significant way), it's better to keep the code simple.